DBAgenda.php

```diff
@@ -0,0 +1,317 @@
+        ));
+        
+        if(isset($vCalendar->VEVENT->ATTENDEE)) {
+            foreach($vCalendar->VEVENT->ATTENDEE as $attendee) {
+                $mail = str_replace("mailto:", "", (string)$attendee);
+
+                $query = $this->pdo->prepare("SELECT * FROM attendees WHERE email=:email;");
+                $query->execute(array(
+                    'email' => $mail
+                ));
+
+                if(is_array($ret = $query->fetch())) {
+                    $this->log->debug("attendee: $mail already exists.");
+                } else {
+                    $user = $this->api->users_lookupByEmail($mail);
+                    if(!is_null($user)) {
+                        $userid = $user->id;
+                    } else {
+                        $userid = null;
+                    }
+
+                    $query = $this->pdo->prepare("REPLACE INTO attendees (email, userid) VALUES (:email, :userid)");
+                    $query->execute(array(
+                        'email' =>  $mail,
+                        'userid' =>  $userid
+                    ));
+                }
+                
+                $query = $this->pdo->prepare("INSERT INTO events_attendees (vCalendarFilename, email) VALUES (:vCalendarFilename, :email)");
+                $query->execute(array(
+                    'vCalendarFilename' =>  $event['vCalendarFilename'],
+                    'email' =>  $mail
+                ));
+            }
+        }
+        
+        if(isset($vCalendar->VEVENT->CATEGORIES)) {
+            foreach($vCalendar->VEVENT->CATEGORIES as $category) {
+                $category = (string)$category;
+                
+                if(is_number_of_attendee_category($category)) {
+                    continue;
+                }
+                
+                $query = $this->pdo->prepare("SELECT * FROM categories WHERE name=:name;");
+                $query->execute(array(
+                    'name' => $category
+                ));
+                
+                $id = null;
+                if(is_array($ret = $query->fetch())) {
+                    $id = $ret['id'];
+                    $this->log->debug("category: $category already exists.");
+                } else {                    
+                    $query = $this->pdo->prepare("INSERT INTO categories (name) VALUES (:name);");
+                    try {
+                        $query->execute(array(
+                            'name' => $category
+                        ));
+                    } catch (PDOException $e) {
+                        if($e->errorInfo[0] === 23000 and $e->errorInfo[1] === 1062) { // Integrity constraint violation: 1062 Duplicate entry
+                            $this->log->warning($e->getMessage());
+                        } else {
+                            throw $e;
+                        }
+                    }
+                    $id = $this->getLastInsertedRowId();
```

Comment:
Will this work as expected in the case where we caught an exception? (I mean: in this case we failed to insert anything with this connection). Could it be safer to re-run `INSERT INTO categories (name) VALUES (:name);`?